### ISSUE TITLE
[Documentation] WSL support for CBL-Mariner?

### ISSUE THREAD

**Issue by rubyFeedback** [microsoft/CBL-Mariner#1159]
Hey guys,

I lately read about CBL-Mariner on reddit. I am using WSL1 (still) on my Win10 laptop;
my main desktop here is on Linux. I am using ubuntu on Win10 but perhaps CBL-Mariner
may be an alternative.

Which brings me to my question: is CBL-Mariner available for installation?

If so, either way, would you guys briefly clarify this on the main README? Perhaps it
may change at a later time, so we could look at the README and then it could have
a sentence like "CBL-Mariner is available via the MS Store" or something like that.
Or that it is presently not available. This mostly just so that people know whether
they could test CBL Mariner rather than, say, Ubuntu instead. Perhaps CBL Mariner
may work better on Windows for instance. (I am still using WSL1 but I suppose
in the long run people will all move to WSL2 so perhaps there are some additional
benefits for CBL Mariner in a WSL2 setup. But I really don't know - it is a genuine
question and mostly focusing on the README.md really)

**Comment by PawelWMS** [comment]
Hey there,

Thanks for showing interest in CBL-Mariner - it's amazing to see it! Unfortunately, CBL-Mariner is an internal Linux distribution and it's not available through any means for 3<sup>rd</sup> party usage. I believe that information is already mentioned at the beginning of this repo's `README.md` but we might need to reword it to avoid any misunderstanding.

I understand your main point was about the contents of `README.md`, but if you'd still like to play around with CBL-Mariner, you could build your own image using the instructions from the [CBL-MarinerDemo repo](https://github.com/microsoft/CBL-MarinerDemo). It has got instructions on [building your own ISO image](https://github.com/microsoft/CBL-MarinerDemo#build-demo-iso) as well, so you can install the system on a physical device.

**Comment by HannahVernon** [comment]
I'd like to be able to build CBL-Mariner in Ubuntu on WSL, however safechroot aborts during creation of the iso.  (I haven't tried the full build).  I imagine that's related to some of the limitations of WSL?  This is the error:

```
ERRO[0000] Mount failed on (/home/hannah/git-repos/CBL-Mariner/toolkit/tools/internal/safechroot/_tmp/TestInitializeShouldCreateRoot/dev). Error: no such file or directory
WARN[0000] Error creating mountpoints for chroot
--- FAIL: TestInitializeShouldCreateRoot (0.00s)
    safechroot_test.go:62:
                Error Trace:    safechroot_test.go:62
                Error:          Received unexpected error:
                                no such file or directory
                Test:           TestInitializeShouldCreateRoot
    safechroot_test.go:71:
                Error Trace:    safechroot_test.go:71
                Error:          Should be true
                Test:           TestInitializeShouldCreateRoot
panic: runtime error: makeslice: cap out of range [recovered]
        panic: runtime error: makeslice: cap out of range
```

**Comment by ohault** [comment]
Does it possible to import CBL-Mariner to use with WSL ( https://docs.microsoft.com/en-us/windows/wsl/use-custom-distro ) 

Could CBL-Mariner be added to the list of available Linux distributions available for download through the online store/Microsoft Store ?

**Comment by Homura37** [comment]
WSL now includes a CBL-Mariner by default.
![QQ截图20211220204825](https://user-images.githubusercontent.com/16538800/146769529-0039c5f0-42e3-434c-a5e9-d7e378a6da80.png)

**Comment by ohault** [comment]
AFIAK, the question is about to the availability of Mariner in the list of supported distribution for WSL, including from the online store : wsl --list --online

**Comment by christopherco** [comment]
Mariner does not officially support WSL at this time. We will continue to re-evaluate this going forward.

**Comment by ohault** [comment]
Hi @christopherco, can you plz don't close this ticket as Mariner still not currently supports this?

**Comment by beedell-roke** [comment]
> https://github.com/microsoft/CBL-Mariner/issues/1159#issuecomment-1841671060

@christopherco, if you are to continue to evaluate this, why close it, especially as [`reason:completed`](https://github.com/microsoft/CBL-Mariner/issues?q=is%3Aissue+reason%3Acompleted)?

**Comment by ohault** [comment]
It's perhaps time for an update on this, especially in the perspective of Azure Linux 3.0 ?
